torch/signal/windows/windows.py

```diff
@@ -0,0 +1,245 @@
+                    dtype: _dtype = None,
+                    layout: _layout = torch.strided,
+                    device: _device = None,
+                    requires_grad: bool = False) -> Tensor:
+    r"""Computes a window with a gaussian waveform.
+
+    The gaussian window is defined as follows:
+
+    .. math::
+        w(n) = w(n) = e^{-\left(\frac{n}{2\sigma}\right)^2}
+
+    Args:
+        window_length (int): the length of the output window.
+            In other words, the number of points of the cosine window.
+        periodic (bool, optional): If `True`, returns a periodic window suitable for use in spectral analysis.
+            If `False`, returns a symmetric window suitable for use in filter design.
+        std (float, optional): the standard deviation of the gaussian. It controls how narrow or wide the window is.
+
+    Keyword args:
+        dtype (:class:`torch.dtype`, optional): the desired data type of returned tensor.
+            Default: if ``None``, uses a global default (see :func:`torch.set_default_tensor_type`).
+        layout (:class:`torch.layout`, optional): the desired layout of returned window tensor.
+            Only `torch.strided` (dense layout) is supported.
+        device (:class:`torch.device`, optional): the desired device of returned tensor.
+            Default: if ``None``, uses the current device for the default tensor type
+            (see :func:`torch.set_default_tensor_type`).
+            :attr:`device` will be the CPU for CPU tensor types and the current CUDA device for CUDA tensor types.
+        requires_grad (bool, optional): If autograd should record operations on the returned tensor. Default: ``False``.
+
+    Returns:
+        (torch.Tensor): window in the form of a tensor.
+
+    Examples:
+        >>> # Generate a gaussian window without keyword args.
+        >>> torch.signal.windows.gaussian_window(10)
+        tensor([1.9287e-22, 1.2664e-14, 1.5230e-08, 3.3546e-04, 1.3534e-01, 1.0000e+00,
+        1.3534e-01, 3.3546e-04, 1.5230e-08, 1.2664e-14])
+
+        >>> # Generate a symmetric gaussian window and standard deviation equal to 0.9.
+        >>> torch.signal.windows.gaussian_window(10, std=0.9, periodic=False)
+        tensor([3.7267e-06, 5.1998e-04, 2.1110e-02, 2.4935e-01, 8.5700e-01, 8.5700e-01,
+        2.4935e-01, 2.1110e-02, 5.1998e-04, 3.7267e-06])
+    """
+    if dtype is None:
+        dtype = torch.get_default_dtype()
+
+    _window_function_checks('cosine_window', window_length, dtype, layout)
+
+    if window_length == 0:
+        return torch.empty((0,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    if window_length == 1:
+        return torch.ones((1,), dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+
+    if periodic:
+        window_length += 1
+
+    k = torch.arange(window_length, dtype=dtype, layout=layout, device=device, requires_grad=requires_grad)
+    k = k - (window_length - 1.0) / 2.0
```

Comment:
With the suggestion for the periodic case, you can simply remove the `if periodic` above and do here
```suggestion
   if not periodic:
     window_lenght -= 1
    k = k - 0.5 * window_length
```

Even more, you can fuse this directly into the `arange` call, as this is just a shift of the whole array, which can be achieved setting `start` and `end` in `arange`.

Reply:
To do this we need to always do `window_length += 1` in case `periodic is True`. Then, I could add the `if not periodic` suggestion, and do some magic with `torch.arange` to fuse it up.